### ISSUE TITLE
[yoshi@next] upgraded and get warnings

### ISSUE THREAD

**Issue by ipanasenko** [wix/yoshi#179]
Hi! Trying out `yoshi@next`, and receive following warnings when running `npm start`:

```
WARNING in ./utils/hoc/index.ts
11:0-110 "export 'IWithRootFolderNameInner' was not found in './withRootFolderName'
 @ ./utils/hoc/index.ts
 @ ./containers/Root/Root.tsx
 @ ./App.tsx
 @ multi ../node_modules/webpack-hot-middleware/client.js?dynamicPublicPath=true&path=__webpack_hmr ./App.tsx

WARNING in ./utils/hoc/index.ts
11:0-110 "export 'IWithRootFolderNameOuter' was not found in './withRootFolderName'
 @ ./utils/hoc/index.ts
 @ ./containers/Root/Root.tsx
 @ ./App.tsx
 @ multi ../node_modules/webpack-hot-middleware/client.js?dynamicPublicPath=true&path=__webpack_hmr ./App.tsx

.....
```

As I see, it's mostly interfaces and types, might be webpack+typescript related?

Branch:
https://github.com/wix-private/media-manager-g6/tree/feature/yoshi-2

**Comment by yairhaimo** [comment]
@ipanasenko - was `ts-loader` patched? Did it fix the issue?

**Comment by ipanasenko** [comment]
@yairhaimo I don't know, I don't follow ts-loader. I will check with latest yoshi version soon

**Comment by ipanasenko** [comment]
@yairhaimo I still get same warnings

**Comment by yairhaimo** [comment]
Ok, will follow up

**Comment by muteor** [comment]
Tried updating ts-loader to:

```
+ yoshi-style-dependencies@3.0.0
+ ts-loader@4.4.2
+ yoshi@3.0.0
```

Still get the warnings.

**Comment by ipanasenko** [comment]
@maxaon helped us figure out why we get these warnings. Turned out, it's because we re-export types (interfaces, enums, etc..) from index.ts files.

We fixed our warnings by removing index.ts files and importing types directly

```diff
- import { MediaType, Layout, Sort } from '../constants';
+ import { MediaType } from '../constants/MediaType';
+ import { Layout } from '../constants/Layout';
+ import { Sort } from '../constants/Sort';
```

**Comment by ipanasenko** [comment]
@muteor I was able to migrate in one working day. Not that scary, as it turned out :)

**Comment by muteor** [comment]
@ranyitz @yairhaimo could we add an option to skip the forcing of esnext? We are really close to our deadline now and I cant spare anyone to do the refactoring, we really need yoshi3 too because CSS ordering is broken in 2.

My only other option is to patch yoshi postinstall or go back to v1, both I dont really want to do :)

**Comment by yairhaimo** [comment]
Other than the warnings in the console, what is the problem itself?
Does the application not work?

Can you check locally the root of the problem?
Try changing this line in your locally installed yoshi:
https://github.com/wix/yoshi/blob/e81bf0e1f2a86d144b4224a2bc862cccaaa14ab5/packages/yoshi/src/loaders/typescript.js#L7

or remove this line:
https://github.com/wix/yoshi/blob/e81bf0e1f2a86d144b4224a2bc862cccaaa14ab5/packages/yoshi/src/loaders/typescript.js#L25

**Comment by muteor** [comment]
The app works but any real warnings are hidden behind all the false-positive warnings.

Its definitely the esnext setting, I have tried removing the threadLoader and that makes no difference.

**Comment by muteor** [comment]
Another way would be to suppress the errors: https://github.com/TypeStrong/ts-loader#transpileonly-boolean-defaultfalse

But for that we would need access to the webpack build to add the filter as I doubt you would want that as the default.

**Comment by yairhaimo** [comment]
Are the warnings false warnings or are they real ones and you just dont have the time right now to deal with them?

**Comment by muteor** [comment]
They are false warnings, they happen when you re-export types.

Example:

```
WARNING in ./src/SiteHeaderMobile/UserMenu/index.ts 1:0-52
"export 'UserMenuProps' was not found in './UserMenu'
 @ ./src/SiteHeaderMobile/SiteHeaderMobile.tsx
 @ ./src/SiteHeaderMobile/SiteHeaderMobile.container.ts
 @ ./src/SiteHeaderMobile/index.ts
 @ ./stories/da-shared-react/SiteHeader/index.tsx
 @ ./stories/da-shared-react/index.tsx
 @ ./stories/index.tsx
 @ ./.storybook/config.js
 @ multi ./node_modules/@storybook/core/dist/server/config/polyfills.js ./node_modules/@storybook/core/dist/server/config/globals.js ./.storybook/config.js (webpack)-hot-middleware/client.js?reload=true
```

https://github.com/wix-private/da-frontend/blob/master/projects/da-shared-react/src/SiteHeaderMobile/UserMenu/index.ts

```
export { default, UserMenuProps } from './UserMenu';
```

So we export the props of the react component, so you can import it like `import { Props } from './package'`

Webpack then sees the export and raises warning because it doesn't know its not a real module. 

I am not sure who is wrong here, its either typescript, webpack, or tsloader but all seem to not care much to actually fix it :)

**Comment by yairhaimo** [comment]
Ok, we will suppress the warnings right now and see if we can find a better solution for the long term.

**Comment by muteor** [comment]
Thanks!

**Comment by ranyitz** [comment]
Released on https://github.com/wix/yoshi/blob/master/CHANGELOG.md#320-jul-25-2018, @muteor can you please verify?

**Comment by muteor** [comment]
@ranyitz I dont see 3.2.0 in the registry?

```
npm view yoshi versions | grep 3.2
  '1.1.302',
  '1.1.312',
  '1.1.322',
  '1.1.332',
```

**Comment by ranyitz** [comment]
@muteor I'm not sure it was published at the time of the comment but I've checked now and it's there, sorry for the trouble.

**Comment by muteor** [comment]
@ranyitz OK got it now but for some reason the filter isn't working...

```WARNING in ./SiteHeaderMobile/UserMenu/index.ts 1:0-52
"export 'UserMenuProps' was not found in './UserMenu'
 @ ./SiteHeaderMobile/SiteHeaderMobile.tsx
 @ ./SiteHeaderMobile/SiteHeaderMobile.container.ts
 @ ./SiteHeaderMobile/index.ts
 @ ./index.ts
```

I see the new code:
```grep -R 'was not found in' node_modules/yoshi
node_modules/yoshi/src/tasks/cdn/utils.js:        warningsFilter: /export .* was not found in/,```

**Comment by yairhaimo** [comment]
ill clone your project and check it out

**Comment by yairhaimo** [comment]
The problem was that i added suppression only for `start` and not for `build`.
I'll send another PR and it will be in the next version.

**Comment by muteor** [comment]
@yairhaimo let me know when this lands and I will retest

**Comment by ranyitz** [comment]
@muteor Landed on [3.3.0](https://github.com/wix/yoshi/blob/master/CHANGELOG.md#nail_care-enhancement)
